src/cert_provider/main.cc

```diff
@@ -383,157 +384,196 @@ int main(int argc, char* argv[]) {
     return 1;
   }
 
-  boost::filesystem::path directory = "/var/sota/import";
-  BasedPath pkey_file = BasedPath("pkey.pem");
-  BasedPath cert_file = BasedPath("client.pem");
-  BasedPath ca_file = BasedPath("root.crt");
-  BasedPath url_file = BasedPath("gateway.url");
-  if (!config_path.empty()) {
-    Config config(config_path);
-    // TODO: provide path to root directory in `--local` parameter
-
-    // try first import base path and then storage path
-    if (!config.import.base_path.empty()) {
-      directory = config.import.base_path;
-    } else if (!config.storage.path.empty()) {
-      directory = config.storage.path;
-    }
+  boost::filesystem::path credentials_path = "";
+  if (commandline_map.count("credentials") != 0) {
+    credentials_path = commandline_map["credentials"].as<boost::filesystem::path>();
+  }
 
-    if (!config.import.tls_pkey_path.empty()) {
-      pkey_file = config.import.tls_pkey_path;
-    } else {
-      pkey_file = config.storage.tls_pkey_path;
-    }
+  if (local_dir.empty() && target.empty()) {
+    std::cerr << "Both a local dir and a target are not specified, thus the resultant cert and key will be lost"
```

Comment:
Good thing to check for. However, I think it's fine to just say "Please provide a local directory and/or target to output the generated files to."

Reply:
ok.